addons/armor_modifier_ace/functions/fnc_handleDamage.sqf

```diff
@@ -0,0 +1,258 @@
+    _oldDamage
+};
+
+// Get scaled armor value of hitpoint and calculate damage before armor
+// We scale using passThrough to handle explosive-resistant armor properly (#9063)
+// We need realDamage to determine which limb was hit correctly
+[_unit, _hitpoint] call ace_medical_engine_fnc_getHitpointArmor params ["_armor", "_armorScaled"];
+private _realDamage = _newDamage * _armor;
+if (!_structuralDamage) then {
+    private _armorCoef = _armor/_armorScaled;
+    private _damageCoef = linearConversion [0, 1, ace_medical_engine_damagePassThroughEffect, 1, _armorCoef];
+    _newDamage = _newDamage * _damageCoef;
+};
+TRACE_6("Received hit",_hitpoint,_ammo,_newDamage,_realDamage,_directHit,_context);
+
+// Drowning doesn't fire the EH for each hitpoint so the "ace_hdbracket" code never runs
+// Damage occurs in consistent increments
+if (
+    _structuralDamage &&
+    {getOxygenRemaining _unit <= 0.5} &&
+    {_damage isEqualTo (_oldDamage + 0.005)}
+) exitWith {
+    TRACE_5("Drowning",_unit,_shooter,_instigator,_damage,_newDamage);
+    ["ace_medical_woundReceived", [_unit, [[_newDamage, "Body", _newDamage]], _unit, "drowning"]] call CBA_fnc_localEvent;
+
+    0
+};
+
+// Faster than (vehicle _unit), also handles dead units
+private _vehicle = objectParent _unit;
+private _inVehicle = !isNull _vehicle;
+private _environmentDamage = _ammo == "";
+
+// Crashing a vehicle doesn't fire the EH for each hitpoint so the "ace_hdbracket" code never runs
+// It does fire the EH multiple times, but this seems to scale with the intensity of the crash
+if (
+    ace_medical_enableVehicleCrashes &&
+    {_environmentDamage && _inVehicle && _structuralDamage} &&
+    {vectorMagnitude (velocity _vehicle) > 5}
+    // todo: no way to detect if stationary and another vehicle hits you
+) exitWith {
+    TRACE_5("Crash",_unit,_shooter,_instigator,_damage,_newDamage);
+    ["ace_medical_woundReceived", [_unit, [[_newDamage, _hitPoint, _newDamage]], _unit, "vehiclecrash"]] call CBA_fnc_localEvent;
+
+    0
+};
+
+// Receiving explosive damage inside a vehicle doesn't trigger for each hitpoint
+// This is the case for mines, explosives, artillery, and catasthrophic vehicle explosions
+if (
+    (!_environmentDamage && _inVehicle && _structuralDamage) &&
+    {
+        private _ammoCfg = configFile >> "CfgAmmo" >> _ammo;
+        GET_NUMBER(_ammoCfg >> "explosive",0) > 0 ||
+        {GET_NUMBER(_ammoCfg >> "indirectHit",0) > 0}
+    }
+) exitWith {
+    TRACE_5("Vehicle hit",_unit,_shooter,_instigator,_damage,_newDamage);
+
+    _unit setVariable ["ace_medical_lastDamageSource", _shooter];
+    _unit setVariable ["ace_medical_lastInstigator", _instigator];
+
+    ["ace_medical_woundReceived", [_unit, [[_newDamage, _hitPoint, _newDamage]], _shooter, "vehiclehit"]] call CBA_fnc_localEvent;
+
+    0
+};
+
+// Get setting for particular unit
+// I don't think we truely need 3 methods avaiable, I would rather choose two 2
```

Comment:
is this comment from you or the source? if the former, what do you mean?

Reply:
Used to be 3 methods, now it's just 2. forgot to update it